src/APL/vercelAPL.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable class-methods-use-this */
+import debugPkg from "debug";
+import fetch from "node-fetch";
+
+import { APL, AuthData } from "./apl";
+
+const debug = debugPkg.debug("VercelAPL");
+
+export const TOKEN_VARIABLE_NAME = "SALEOR_AUTH_TOKEN";
+export const DOMAIN_VARIABLE_NAME = "SALEOR_DOMAIN";
+export const SALEOR_REGISTER_APP_URL = "SALEOR_REGISTER_APP_URL";
+export const SALEOR_DEPLOYMENT_TOKEN = "SALEOR_DEPLOYMENT_TOKEN";
+
+const envAuthData = (): AuthData | undefined => {
+  const token = process.env[TOKEN_VARIABLE_NAME];
+  const domain = process.env[DOMAIN_VARIABLE_NAME];
+  if (!token || !domain) {
+    return undefined;
+  }
+  return {
+    token,
+    domain,
+  };
+};
+
+const saveDataToVercel = async (
+  registerAppURL: string,
+  deploymentToken: string,
+  authData?: AuthData
+) => {
+  debug("Saving data to Vercel");
+
+  try {
+    await fetch(registerAppURL, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        token: deploymentToken,
+        envs: {
+          [TOKEN_VARIABLE_NAME]: authData?.token || "",
+          [DOMAIN_VARIABLE_NAME]: authData?.domain || "",
+        },
+      }),
+    });
+  } catch (error) {
+    debug("Error during saving the data:", error);
+  }
+};
+
+export type VercelAPLConfig = {
+  registerAppURL?: string;
+  deploymentToken?: string;
```

Comment:
Can you add JSDoc to these too? It will explain users, can link to docs, will be available in IDE